### DISCUSSION TITLE
Can deferEdit have ability to modify allowedMention as well?

### DISCUSSION THREAD

**Discussion by MandarinSmell** [discord-jda/JDA#2170] [Questions and Help] [answered]
Hello, it's first time to use JDA, just started to use it days ago. I'm trying to perform message with Discord SlectMenu, Button, etc. I made my own class, so that bot can handle component interactions like below

```java
public void performInteraction(GenericComponentInteractionCreateEvent event) {
    MessageChannel ch = event.getChannel();
    Guild g = msg.getGuild();

    //... After some codes

    event.deferEdit()
             .setContent("Text with role ping") //Text is example
             .setActionRows(getComponents(g))
             .queue();

    //... Other codes here
}
```

I made message not ping any roles or members when creating message like below

```java
ch.sendMessage("Text with role ping") //Text is example
         .allowedMneitons(new ArrayList<>())
         .setActionRows(//Add Components)
         .queue();
```

I now have problem when use this code that if I use one of component in this message, bot will change role ping depending on SlectMenu that I added. Problem is that it starts to ping role even though I made it not do that. Actually, the reason why first code block lacks method about allowed mention is because `deferEdit()`, or `MessageEditCallbackAction` doesn't support such. Storing `Message` instance of this message could make me able to edit message with allowed mention modifying, but then component now shows `This interaction failed` (...obviously)

So I wonder if there's way to make my bot not ping (even though editing message doesn't give mentions, but still) role/members when using `deferEdit()` method. I decided to choose `Question` one instead of `Feature Request` because I lack deep knowledge of Discord data structure. Like, it can be that I'm asking this without knowing that deferEdit feature can't support modifying allowedMentions. As long as Discord allows, can deferEdit() also have modifies of allowedMention?

```java
event.deferEdit()
        .setContent("Text with role ping") //Text is example
        .setActionRows(getComponents(g))
        .allowedMentions(new ArrayList<>())
        .queue();
```

**Comment by Splayfery** [comment]
You should use .deferReply() instead of deferEdit()

**Comment by MandarinSmell** [comment]
That creates new message while I want to edit original message. `Message.editMessage()` can provide modifying allowed mention without creating new message, so it should work same for `deferEdit()` way unless Discord really didn't support such

**Comment by MinnDevelopment** [comment, marked as answer]
With the Message Rework (#2187), you should be able to modify allowed mentions on every edit request.
